mantidproject/mantid
# Crystal field uses ifunction object in python interface

Refactored Crystal Field's python interface to use the C++ IFunction object directly for setting and getting function properties. Also fixed some issues concerning setting of constraints and ties on parameters of the crystal field functions. Due to the structure of those functions it was impossible to set some of the ties. I have moved implementation of ties and constraints from `ParamFunction` to `IFunction`.

**To test:**

* Code review
* Check that all tests pass
* Check that the crystal field functionality didn't change.

Fixes #18518.

*Does not need to be in the release notes.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
